.github/workflows/gallery.yml

```diff
@@ -8,21 +8,33 @@ on:
       - master
 permissions:
   contents: write
-
+  id-token: write
+  pages: write
 jobs:
   test:
     name: Publish Gallery to Github Pages
     if: github.repository != 'gautamkrishnar/github-pages-gallery'
     runs-on: ubuntu-latest
     steps:
+      - name: Setup GitHub Pages ⚙️
+        uses: actions/configure-pages@v3.0.4
       - name: Checkout 🛎️
         uses: actions/checkout@v3
         with:
           persist-credentials: false
       - name: Build Gallery 🔧
         run: docker run -v "$(pwd):/work" thumbsupgallery/thumbsup /bin/sh -c "cd /work/ && thumbsup --config config.json"
+      - name: Package and upload artifact 📦
+        uses: actions/upload-pages-artifact@v1.0.7
+        with:
+          path: 'build_output'
+          name: gh-pages
       - name: Deploy to Github Pages 🚀
         uses: actions/deploy-pages@v1
         with:
           artifact_name: gh-pages
           token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Delete artifact 🗑️
+        uses: geekyeggo/delete-artifact@v2
+        with:
+          name: gh-page
```

Comment:
Delete uploaded artifact in order to In order not to be blocked by the limitations of GitHub Storage.

At this time, there is no official GitHub Action for that